### PR TITLE
6219 686 spouse marriage history part II

### PR DESCRIPTION
**Pull request by micahchiang**
## Description
[Original Ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6219)

This pull request adds pages 4 & 5 to the `add spouse` workflow of the 686. Specifically, it allows a user to report any past marriages their spouse may have had in keeping with the requirements found on the paper version of the 686. 

Unit tests will be added in a separate pull request.

## Testing done
- local

## Screenshots
Screen Shot 2020-03-05 at 9 34 55 AM
Screen Shot 2020-03-05 at 9 43 50 AM
Screen Shot 2020-03-05 at 9 50 25 AM
Screen Shot 2020-03-05 at 9 50 48 AM
Screen Shot 2020-03-05 at 9 51 28 AM

## Acceptance criteria
- [x] pages have been added.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by cohnjesse — APPROVED** [review]
LGTM

File: src/applications/disability-benefits/new-686/config/chapters/report-add-a-spouse/current-spouse-marriage-history-details/currentSpouseMarriageHistoryDetails.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+};
+
+export const uiSchema = {
+  spouseMarriageHistory: {
+    items: {
+      'ui:title': spouseTitle,
+      marriageStartDate: {
+        ...currentOrPastDateUI('Date of marriage'),
+        ...{
+          'ui:required': formData => formData.spouseWasMarriedBefore === true,
```
Comment (billfienberg) [inline src/applications/disability-benefits/new-686/config/chapters/report-add-a-spouse/current-spouse-marriage-history-details/currentSpouseMarriageHistoryDetails.js]: NIT: If `spouseWasMarriedBefore` is always boolean, is it necessary to compare it to `true`? 

Would either of the following work?
```
// if `spouseWasMarriedBefore` is guaranteed to be a boolean, then return the boolean
- 'ui:required': formData => formData.spouseWasMarriedBefore === true,
+ 'ui:required': formData => formData.spouseWasMarriedBefore,
```
or
```
// if `spouseWasMarriedBefore` could be `null` or `undefined`, then coerce the truthiness of it
- 'ui:required': formData => formData.spouseWasMarriedBefore === true,
+ 'ui:required': formData => !!formData.spouseWasMarriedBefore,
```

File: src/applications/disability-benefits/new-686/config/chapters/report-add-a-spouse/current-spouse-marriage-history-details/helpers.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+export function spouseTitle({ formData }) {
```
Comment (billfienberg) [inline src/applications/disability-benefits/new-686/config/chapters/report-add-a-spouse/current-spouse-marriage-history-details/helpers.jsx]: NIT: According to the React [docs](https://reactjs.org/docs/jsx-in-depth.html#user-defined-components-must-be-capitalized), I think this is supposed to be capitalized.

```
- export function spouseTitle({ formData }) {
+ export function SpouseTitle({ formData }) {
```

**Review by billfienberg — APPROVED** [review]
Ideally, we would have some tests on this code. That being said, it's pretty much all new code, and looks reasonably safe.

File: src/applications/disability-benefits/new-686/config/chapters/report-add-a-spouse/current-spouse-marriage-history-details/currentSpouseMarriageHistoryDetails.js (hunk 1 above)
Comment (micahchiang) [reply src/applications/disability-benefits/new-686/config/chapters/report-add-a-spouse/current-spouse-marriage-history-details/currentSpouseMarriageHistoryDetails.js]: Refactored [here](https://github.com/department-of-veterans-affairs/vets-website/pull/11820/commits/ff8536a2d81259a55660938c1de9296259bda912)

File: src/applications/disability-benefits/new-686/config/chapters/report-add-a-spouse/current-spouse-marriage-history-details/helpers.jsx (hunk 2 above)
Comment (micahchiang) [reply src/applications/disability-benefits/new-686/config/chapters/report-add-a-spouse/current-spouse-marriage-history-details/helpers.jsx]: Refactored [here](https://github.com/department-of-veterans-affairs/vets-website/pull/11820/commits/ff8536a2d81259a55660938c1de9296259bda912)

**Pull request merged**